core/src/main/java/hudson/model/Queue.java

```diff
@@ -433,6 +441,22 @@ public synchronized void load() {
         }
     }
 
+    private long getMaxNextBuildNumber() {
+        long maxNextBuildNumber = 0;
+        Jenkins jenkins = Jenkins.getInstance();
+
+        if (jenkins != null) {
+            for (String jobName : jenkins.getJobNames()) {
+                TopLevelItem item = jenkins.getItem(jobName);
+                if (item instanceof Job) {
+                    maxNextBuildNumber = Math.max(maxNextBuildNumber, ((Job) item).getNextBuildNumber());
```

Comment:
this code is wrong if someone has had more than 2^31-1 builds.

nextBuildNumber can be set manually and a user may have set it ti this riduculous value for some specific reason.
(2^31-4) and had a few builds of the project